src/ert/job_queue/queue.py

```diff
@@ -166,27 +169,37 @@ def max_running(self) -> int:
             return len(self._realizations)
         return max_running
 
-    def available_capacity(self) -> bool:
-        return (
-            self.count_realization_state(RealizationState.RUNNING) < self.max_running()
-        )
-
     def is_all_reals_state(self, state: RealizationState) -> bool:
         return all(real.current_state == state for real in self._realizations)
 
-    async def launch_jobs(self) -> None:
-        while self.available_capacity():
-            try:
-                realization = next(
-                    (
-                        real
-                        for real in self._realizations
-                        if real.current_state == RealizationState.WAITING
-                    )
-                )
-                realization.submit()
-            except StopIteration:
-                break
+    async def _realization_submitter(self) -> None:
+        submit_delta = datetime.timedelta(seconds=self._queue_config.submit_sleep)
+        time_of_last_submission: datetime.datetime = (
+            datetime.datetime.now() - submit_delta
+        )
+        while self.is_active():
+            while (
+                self.count_realization_state(RealizationState.RUNNING)
+                + self.count_realization_state(RealizationState.SUBMITTED)
+                < self.max_running()
+            ):
+                if datetime.datetime.now() - time_of_last_submission >= submit_delta:
```

Comment:
I thought that this check is not required if we do `await asyncio.sleep(self._queue_config.submit_sleep)` or?

Reply:
removed now